src/sentry/migrations/0252_add_widget_type.py

```diff
@@ -0,0 +1,39 @@
+# Generated by Django 2.2.24 on 2021-11-19 03:32
+
+from django.db import migrations
+
+import sentry.db.models.fields.bounded
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0251_sentryappavatar_sentryapp_not_unique"),
+    ]
+
+    operations = [
+        migrations.AddField(
+            model_name="dashboardwidget",
+            name="widget_type",
+            field=sentry.db.models.fields.bounded.BoundedPositiveIntegerField(default=0),
```

Comment:
Adding columns with default values will take out locks on the table which will block reads and writes. We generally recommend that you add nullable columns without a default, and then use a second migration to set the default value and apply the not null constraint in two steps. The [migration authoring docs](https://develop.sentry.dev/database-migrations/#adding-not-null-to-columns) have an example of how this would work.